Library/Homebrew/cleaner.rb

```diff
@@ -88,10 +88,14 @@ def clean_dir(d)
 
       Find.prune if @f.skip_clean? path
 
-      next if path.symlink? || path.directory?
+      next if path.directory?
 
       if path.extname == ".la"
         path.unlink
+      elsif path.symlink? && path.exist?
+        if path.readlink.extname == ".la"
+          path.unlink
+        end
```

Comment:
I suggest…
```ruby
      elsif path.symlink?
        path.unlink if path.exist? && path.realpath.extname == ".la"
```
This code handles the case where the symlink does not have a `.la` extension, and that symlink links to a file that does have a `.la` extension. Do you have an example of that case? If not, I suggest the simpler…
```ruby
      elsif path.symlink?
        # Skip symlinks
```
The symlinks with `.la` extension will be removed by line 93 `if path.extname == ".la"`.

Reply:
@sjackman:  True, this is likely a case of 'YAGNI.'